Content.Server/Afk/AFKSystem.cs

```diff
@@ -77,16 +89,29 @@ public override void Update(float frameTime)
         {
             var pSession = (IPlayerSession) session;
             var isAfk = _afkManager.IsAfk(pSession);
-
-            if (isAfk && _afkPlayers.Add(pSession))
+            if (isAfk)
             {
-                var ev = new AFKEvent(pSession);
-                RaiseLocalEvent(ref ev);
-                continue;
+                var curTime = _gameTiming.CurTime.TotalSeconds;
+                if (!_afkPlayers.ContainsKey(pSession))
+                {
+                    _afkPlayers.Add(pSession, curTime);
+                    var ev = new AFKEvent(pSession);
+                    RaiseLocalEvent(ref ev);
+                    continue;
+                }
+                var timeElapsed = curTime - _afkPlayers[pSession];
+                if (timeElapsed > _kickDelay)
+                {
+                    _netManager.DisconnectChannel(pSession.ConnectedClient, Loc.GetString("afk-system-kick-reason"));
+                }
+                else if(timeElapsed >= _kickDelay-1)
```

Comment:
isnt this going to give the warning 1 second before you're kicked?

Reply:
Also it's going to send it every time the afk system runs; if a downstream has this at 1 second intervals the message is gonna get spammed a lot.